### PR TITLE
feat: DAH-2387: Adaptive password validation

### PR DESCRIPTION
**Pull request by chadbrokaw**
[DAH-2387](https://sfgovdt.jira.com/browse/DAH-2387)

[DAH-2387]: https://sfgovdt.jira.com/browse/DAH-2387?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/api/authApiService.ts [hunk 1]
```diff
@@ -32,16 +33,16 @@ export const forgotPassword = async (email: string): Promise<string> =>
...
-  token: string,
-  password: string,
-  passwordConfirmation: string
-): Promise<string> =>
-  put<{ accessToken: string }>("/user/update-password", {
-    password: password,
-    passwordConfirmation: passwordConfirmation,
-    token: token,
-  }).then(({ data }) => data.accessToken)
+// export const updatePassword = async (
```
Comment (tallulahkay) [inline app/javascript/api/authApiService.ts:36]: you can just delete this!

File: app/javascript/pages/account/PasswordFieldset.tsx [hunk 2]
```diff
@@ -1,22 +1,91 @@
-import { Field, passwordRegex, t } from "@bloom-housing/ui-components"
+import { Field, FieldProps, passwordRegex, t } from "@bloom-housing/ui-components"
 import React from "react"
 import { UseFormMethods } from "react-hook-form"
 import "./account-settings"
 import Fieldset from "./Fieldset"
+import { Icon } from "@bloom-housing/ui-seeds"
+import { CheckIcon, EyeIcon, EyeSlashIcon, XMarkIcon } from "@heroicons/react/24/outline"
+
+const instructionListItem = (
```
Comment (tallulahkay) [inline app/javascript/pages/account/PasswordFieldset.tsx:9]: can you capitalize the beginning to match the patterns of the rest of the file

**Review by tallulahkay — CHANGES_REQUESTED** [review]
Looks rly good!! Two updates needed.
1. Validation state needs to reset after the password is submitted successfully 
Screenshot 2024-08-07 at 2 58 37 PM

2. Validation state needs to be reset after the password is submitted but the old password doesn't match. (Do we need to show an adaptive error message saying old password doesn't match?)
Screenshot 2024-08-07 at 2 56 30 PM

**Comment by chadbrokaw** [comment]
> Looks rly good!! Two updates needed.
> 
> 1. Validation state needs to reset after the password is submitted successfully
> 
> Screenshot 2024-08-07 at 2 58 37 PM
> 2. Validation state needs to be reset after the password is submitted but the old password doesn't match. (Do we need to show an adaptive error message saying old password doesn't match?)
> 
> Screenshot 2024-08-07 at 2 56 30 PM

Thank you for catching that! The validation should clear on submit now

**Review by tallulahkay — APPROVED** [review]
Amazing work!

**Review by tallulahkay — APPROVED** [review]

**Pull request merged**